src/Agent.Worker/JobExtensionRunner.cs

```diff
@@ -10,10 +10,10 @@ namespace Microsoft.VisualStudio.Services.Agent.Worker
 {
     public sealed class JobExtensionRunner : IStep
     {
-        private readonly Func<Task<TaskResult>> _runAsync;
+        private readonly Func<Task> _runAsync;
```

Comment:
I removed TaskResult from the return type. It already lives on the execution context which is available to the job extension.